src/content/SectionPopup/GradesComponent/GradesComponent.tsx

```diff
@@ -0,0 +1,68 @@
+import { useEffect, useState } from "react";
+import { IGradesAPIData, getGradesData, getGradesUrl } from "./GradesHelper";
+import { ISectionData } from "../../App/App.types";
+import "./GradesComponent.css";
+
+interface IProps {
+  selectedSection: ISectionData;
+}
+
+const GradesComponent = ({ selectedSection }: IProps) => {
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+  const [gradesData, setGradesData] = useState<IGradesAPIData>();
+  const [isError, setIsError] = useState<boolean>(false);
+
+  useEffect(() => {
+    const fetchGrades = async () => {
+      setIsLoading(true);
+      setIsError(false);
+      try {
+        let data = await getGradesData(selectedSection);
+        setGradesData(data);
+      } catch (error) {
+        setIsError(true);
+      } finally {
+        setIsLoading(false);
+      }
+    };
+    fetchGrades();
+  }, [selectedSection]);
+
+  const gradesURL = getGradesUrl(selectedSection);
+
+  return (
+    <div className="GradesComponentContainer">
+      <div className="GradesInformationTitle">Grade Information: </div>
+      {isError ? (
+        <div>Error loading grades.</div>
+      ) : isLoading ? (
+        <div>Loading...</div>
+      ) : (
+        <>
+          <div className="GradesContainer">
+            <table>
+              <tr>
+                <td>Average (All Time)</td>
+                <td>{gradesData?.average.toFixed(2)}</td>
+              </tr>
+              <tr>
+                <td>Average (5 Years)</td>
+                <td>{gradesData?.averageFiveYears.toFixed(2)}</td>
+              </tr>
```

Comment:
Clicking sections without a five year average like AFST_V 350-A_101 breaks the extension, just shows a white page. Maybe show whatever's available:
```suggestion
              {gradesData?.average && <tr>
                <td>Average (All Time)</td>
                <td>{gradesData?.average.toFixed(2)}</td>
              </tr>}
              {gradesData?.averageFiveYears && <tr>
                <td>Average (5 Years)</td>
                <td>{gradesData?.averageFiveYears.toFixed(2)}</td>
              </tr>}
```

Reply:
will add null checks good catch

Reply:
@Rileyc2022 Pushed, can you try? 
![image](https://github.com/mlool/workday-calendar-extension/assets/89222842/94168694-cba9-4432-8a85-994ca5516b9a)

Reply:
yeah tested lgtm